packages/@react-spectrum/table/test/TreeGridTable.test.tsx

```diff
@@ -0,0 +1,1518 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+jest.mock('@react-aria/live-announcer');
+import {act, fireEvent, installPointerEvent, render as renderComponent, triggerPress, within} from '@react-spectrum/test-utils';
+import {announce} from '@react-aria/live-announcer';
+import {composeStories} from '@storybook/testing-react';
+import {enableTableNestedRows} from '@react-stately/flags';
+import {Provider} from '@react-spectrum/provider';
+import React from 'react';
+import {Scale} from '@react-types/provider';
+import * as stories from '../stories/TreeGridTable.stories';
+import {tableTests} from './Table.test';
+import {theme} from '@react-spectrum/theme-default';
+import userEvent from '@testing-library/user-event';
+
+let {
+  StaticExpandableRows: StaticExpandableTable,
+  DynamicExpandableRowsStory: DynamicExpandableTable,
+  ManyExpandableRowsStory: ManyRowsExpandableTable,
+  EmptyTreeGridStory: EmptyStateTable,
+  LoadingTreeGridStory: LoadingTable,
+  UserSetRowHeader: UserSetRowHeaderTable
+} = composeStories(stories);
+
+let onSelectionChange = jest.fn();
+let onExpandedChange = jest.fn();
+let onAction = jest.fn();
+
+let getCell = (tree, text) => {
+  // Find by text, then go up to the element with the cell role.
+  let el = tree.getByText(text);
+  while (el && !/gridcell|rowheader|columnheader/.test(el.getAttribute('role'))) {
+    el = el.parentElement;
+  }
+
+  return el;
+};
+
+let focusCell = (tree, text) => act(() => getCell(tree, text).focus());
+let moveFocus = (key, opts = {}) => {
+  fireEvent.keyDown(document.activeElement, {key, ...opts});
+  fireEvent.keyUp(document.activeElement, {key, ...opts});
+};
+
+let render = (children, scale = 'medium' as Scale, locale = 'en-US') => {
+  let tree = renderComponent(
+    <Provider theme={theme} scale={scale} locale={locale}>
+      {children}
+    </Provider>
+  );
+
+  act(() => {jest.runAllTimers();});
+  return tree;
+};
+
+let rerender = (tree, children, scale = 'medium' as Scale) => {
+  let newTree = tree.rerender(
+    <Provider theme={theme} scale={scale}>
+      {children}
+    </Provider>
+  );
+  act(() => {jest.runAllTimers();});
+  return newTree;
+};
+
+enableTableNestedRows();
+
+describe.only('TableView tests with expandable rows flag on', tableTests);
+
+describe.only('TableView with expandable rows', function () {
```

Comment:
these `.only` make it so the table tests don't run an extra time due to the way jest imports work.

Reply:
> Another way would be to move the tests out into a separate util maybe

Could you elaborate on this a bit? Definitely interested in any way to get rid of the `.only`'s but not sure I understand what that util would be

Reply:
I meant pulling out the `tableTests` function into a separate file, and importing it from both `Table.test.js` and `TreeGridTable.test.tsx` so that it doesn't have a `describe` in it. Then you shouldn't need the only?

Reply:
Ah, good idea. Tried it out and it works, but opted to move the 'TableView tests with expandable rows flag on' test suite into Table.test.js instead since we'll be getting rid of it later anyways (not sure why I didn't just do that to start with). I've stashed the "`tableTests` function into separate file" work in case if we still want that, figured that approach is more appropriate if the Table tests were actually generalized enough to test basic table interactions with the TreeGridView render instead.

I may look into refactoring/generalizing those Table tests and moving it into a util function in the future to make it easier to get full coverage when we make the RAC implementation as well.